NestJS/src/gmail_api/gmailapi.js

```diff
@@ -37,130 +29,148 @@ exports.getAllMessages = function(){
  * @param {Object} credentials The authorization client credentials.
  * @param {function} callback The callback to call with the authorized client.
  */
-async function authorize(callback, data, recipient) {
-  // const {client_secret, client_id, redirect_uris} = credentials.installed;
+
+authorize = async (callback, data, recipient) => {
```

Comment:
you are making this global?

Reply:
Yes, for send message and retrieve messages from gmail we use one authorize function.

Reply:
And for every action, we need to auth.

Reply:
i am not recommending to use global variables since this is hard to track and maintain them, please consider to use import\export instead